From 11e477ddf14428ca960c724409b98cd9cc71829d Mon Sep 17 00:00:00 2001
From: Jeevaka Badrappan <jeevaka.badrappan@intel.com>
Date: Thu, 28 Mar 2013 15:32:27 +0200
Subject: [REVERT ME] Telephony: Add radio off reason for RADIO_POWER request

BZ: 102058

With this patch, radio power off reasons is sent
as part of the RIL_REQUEST_RADIO_POWER request.

Severity: Medium

This patch should be reverted with the original patch
http://android.intel.com:8080/#/c/108568/ when rebasing
to Kitkat official release.

Change-Id: Ibcdd1f52e303a48b91d2b46417e8d03ed0b0013b
Orig-Change-Id: I11a063ee6668ee806e6ba7a0a3a9c2018962471f
Category: device enablement
Domain: Telephony-Modem-Common
Origin: internal
Upstream-Candidate: no.
Signed-off-by: Jeevaka Badrappan <jeevaka.badrappan@intel.com>
---
 src/java/com/android/internal/telephony/RIL.java |   22 +++++++++++++++++++++-
 1 files changed, 21 insertions(+), 1 deletions(-)

diff --git a/src/java/com/android/internal/telephony/RIL.java b/src/java/com/android/internal/telephony/RIL.java
index a227008..fec5b8d 100644
--- a/src/java/com/android/internal/telephony/RIL.java
+++ b/src/java/com/android/internal/telephony/RIL.java
@@ -33,6 +33,7 @@ import android.content.IntentFilter;
 import android.net.ConnectivityManager;
 import android.net.LocalSocket;
 import android.net.LocalSocketAddress;
+import android.provider.Settings;
 import android.os.AsyncResult;
 import android.os.Handler;
 import android.os.HandlerThread;
@@ -1427,9 +1428,28 @@ public final class RIL extends BaseCommands implements CommandsInterface {
     setRadioPower(boolean on, Message result) {
         RILRequest rr = RILRequest.obtain(RIL_REQUEST_RADIO_POWER, result);
 
-        rr.mParcel.writeInt(1);
+        rr.mParcel.writeInt(2);
         rr.mParcel.writeInt(on ? 1 : 0);
 
+        int radioOffReason = 0;//RILConstants.RADIO_OFF_REASON_NONE;
+
+        if (!on) {
+            String rebootOrShutdownRequested =
+                    SystemProperties.get("sys.shutdown.requested", null);
+            if (rebootOrShutdownRequested != null
+                    && rebootOrShutdownRequested.length() > 0) {
+                char reason = rebootOrShutdownRequested.charAt(0);
+                if (reason == '0' || reason == '1') {
+                    radioOffReason = 1;//RILConstants.RADIO_OFF_REASON_SHUTDOWN;
+                }
+            } else if (Settings.Global.getInt(mContext.getContentResolver(),
+                    Settings.Global.AIRPLANE_MODE_ON, 0) != 0) {
+                radioOffReason = 2;//RILConstants.RADIO_OFF_REASON_AIRPLANE_MODE;
+            }
+        }
+
+        rr.mParcel.writeInt(radioOffReason);
+
         if (RILJ_LOGD) {
             riljLog(rr.serialString() + "> " + requestToString(rr.mRequest)
                     + (on ? " on" : " off"));
-- 
1.7.4.1

